src/Tracer.php

```diff
@@ -148,38 +136,58 @@ public static function columnAttributes($column): string
-            'binary' => 'binary',
-            'blob' => 'binary',
-            'boolean' => 'boolean',
+        $type = match ($column['type']) {
+            'tinyint(1)', 'bit' => 'boolean',
+            'nvarchar(max)' => 'text',
+            default => null,
+        };
+
+        $type ??= match ($column['type_name']) {
+            'bigint', 'int8' => 'biginteger',
+            'binary', 'varbinary', 'bytea', 'image', 'blob', 'tinyblob', 'mediumblob', 'longblob' => 'binary',
+            // 'bit', 'varbit' => 'bit',
+            'boolean', 'bool' => 'boolean',
+            'char', 'bpchar', 'nchar' => 'char',
             'date' => 'date',
-            'date_immutable' => 'date',
-            'dateinterval' => 'date',
-            'datetime' => 'datetime',
-            'datetime_immutable' => 'datetime',
-            'datetimetz' => 'datetimetz',
-            'datetimetz_immutable' => 'datetimetz',
-            'decimal' => 'decimal',
+            'datetime', 'datetime2' => 'datetime',
+            'datetimeoffset' => 'datetimetz',
+            'decimal', 'numeric' => 'decimal',
+            'double', 'float8' => 'double',
             'enum' => 'enum',
-            'float' => 'float',
-            'guid' => 'string',
-            'integer' => 'integer',
+            'float', 'real', 'float4' => 'float',
+            'geography' => 'geography',
+            'geometry', 'geometrycollection', 'linestring', 'multilinestring', 'multipoint', 'multipolygon', 'point', 'polygon' => 'geometry',
+            // 'box', 'circle', 'line', 'lseg', 'path' => 'geometry',
+            'integer', 'int', 'int4' => 'integer',
+            // 'inet', 'cidr' => 'ipaddress',
+            // 'interval' => 'interval',
             'json' => 'json',
+            'jsonb' => 'jsonb',
             'longtext' => 'longtext',
-            'object' => 'string',
-            'simple_array' => 'string',
-            'smallint' => 'smallinteger',
-            'string' => 'string',
-            'text' => 'text',
+            // 'macaddr', 'macaddr8' => 'macadress',
+            'mediumint' => 'mediuminteger',
+            'mediumtext' => 'mediumtext',
+            // 'money', 'smallmoney' => 'money',
+            // 'set' => 'set',
+            'smallint', 'int2' => 'smallinteger',
+            'text', 'ntext' => 'text',
+            // 'tsvector', 'tsquery' => 'text',
             'time' => 'time',
-            'time_immutable' => 'time',
-        ];
-
-        return $mappings[$type] ?? 'string';
+            'timestamp' => 'timestamp',
+            'timestamptz' => 'timestamptz',
+            'timetz' => 'timetz',
+            'tinyint' => 'tinyinteger',
+            'tinytext' => 'tinytext',
+            // 'uuid', 'uniqueidentifier' => 'uuid',
+            'varchar', 'nvarchar' => 'string',
+            // 'xml' => 'xml',
+            // 'year' => 'year',
+            default => null,
+        };
+
+        return $type ?? 'string';
```

Comment:
The mapping contains almost all possible column types for `mysql`, `mariadb`, `sqlite`, `pgsql` and `sqlsrv`.